php/class-wp-seo.php

```diff
@@ -251,12 +251,15 @@ public function save_post_fields( $post_id ) {
 	 * Add the meta box to taxonomies with per-term fields enabled.
 	 */
 	public function add_term_boxes() {
-		foreach ( WP_SEO_Settings()->get_enabled_taxonomies() as $slug ) {
-			add_action( $slug . '_add_form_fields', array( $this, 'add_term_meta_fields' ), 10, 2 );
-			add_action( $slug . '_edit_form', array( $this, 'edit_term_meta_fields' ), 10, 2 );
+		$taxonomies = WP_SEO_Settings()->get_enabled_taxonomies();
+		if ( ! empty( $taxonomies ) ) {
+			foreach ( $taxonomies as $slug ) {
+				add_action( $slug . '_add_form_fields', array( $this, 'add_term_meta_fields' ), 10, 2 );
+				add_action( $slug . '_edit_form', array( $this, 'edit_term_meta_fields' ), 10, 2 );
+			}
+			add_action( 'created_term', array( $this, 'save_term_fields' ), 10, 3 );
+			add_action( 'edited_term', array( $this, 'save_term_fields' ), 10, 3 );
```

Comment:
You might need to make a similar change to `WP_SEO_Settings::has_post_fields()` and `WP_SEO_Settings::has_term_fields()`, both of which immediately call `in_array()`. (Alternatively, add a `! empty()` check to `get_enabled_taxonomies()` and `get_enabled_post_types()`.)